File: examples/eAPI v1.8/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/CartItem.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects
{
    public class CartItem : ApiBase
    {
        [JsonProperty("name")]
        public string Name{ get; set; }

        [JsonProperty("quantity")]
        public int Quantity { get; set; }

        [JsonProperty("amount")]
        public long Amount { get; set; }

        [JsonProperty("description")]
        public string Description{ get; set; }

        public CartItem()
        {
        }

        public CartItem(string name, int quantity, long amount)
        {
            this.Name = name;
            this.Quantity = quantity;
            this.Amount = amount;
        }

        public CartItem(string name, int quantity, long amount, string description)
        {
            this.Name = name;
            this.Quantity = quantity;
            this.Amount = amount;
            this.Description = description;
        }

        public string ToSign()
        {
            StringBuilder sb = new StringBuilder();
            Add(sb, Name);
            Add(sb, Quantity);
            Add(sb, Amount);
            Add(sb, Description);
            DeleteLast(sb);
            return sb.ToString();
        }
    }
}

File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/CartItem.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@JsonInclude(value = JsonInclude.Include.NON_NULL)
public class CartItem extends ApiBase implements Signable {

	private static final long serialVersionUID = -3825192932302805075L;

	private String name;

	private int quantity;

	private long amount;

	private String description;


	public CartItem(String name, int quantity, long amount) {
		this.name = name;
		this.quantity = quantity;
		this.amount = amount;
	}

	public CartItem(String name, int quantity, long amount, String description) {
		this(name, quantity, amount);
		this.description = description;
	}

	@Override
	public String toSign() {
		StringBuilder sb = new StringBuilder(); 
		add(sb, name);
		add(sb, quantity);
		add(sb, amount);
		add(sb, description);
		deleteLast(sb);
		return sb.toString();
	}

}
